### PR TITLE
add log line for DeleteWorkflowExecution when request hits HistoryService

### PR DESCRIPTION
**Pull request by lina-temporal**
## What changed?

Adds logging for when the history service recieves the DeleteWorkflowExecution request. Previously, these were logged as part of the retention timer firing from the timer queue processor.

## Why?

Gives us more visibility on DeleteWorkflowExecution up-front, instead of when the task is processed later. 

## How did you test it?

```
$ TEST_ARGS="-v -run TestFunctionalSuite -testify.m TestDeleteWorkflowExecution_CompetedWorkflow" make functional-test
=== RUN   TestFunctionalSuite
2024-06-13T12:33:58.519-0700	info	created database	{"database": "test_0613193358_rge", "logging-call-at": "test.go:182"}
2024-06-13T12:33:59.371-0700	info	loaded schema	{"logging-call-at": "test.go:205"}
=== RUN   TestFunctionalSuite/TestDeleteWorkflowExecution_CompetedWorkflow
==========================================================================
Send commands to server using RespondWorkflowTaskCompleted:
--------------------------------------------------------------------------
command_type:COMMAND_TYPE_COMPLETE_WORKFLOW_EXECUTION  complete_workflow_execution_command_attributes:{}
==========================================================================
Send commands to server using RespondWorkflowTaskCompleted:
--------------------------------------------------------------------------
command_type:COMMAND_TYPE_COMPLETE_WORKFLOW_EXECUTION  complete_workflow_execution_command_attributes:{}
==========================================================================
Send commands to server using RespondWorkflowTaskCompleted:
--------------------------------------------------------------------------
command_type:COMMAND_TYPE_COMPLETE_WORKFLOW_EXECUTION  complete_workflow_execution_command_attributes:{}
==========================================================================
Send commands to server using RespondWorkflowTaskCompleted:
--------------------------------------------------------------------------
command_type:COMMAND_TYPE_COMPLETE_WORKFLOW_EXECUTION  complete_workflow_execution_command_attributes:{}
==========================================================================
Send commands to server using RespondWorkflowTaskCompleted:
--------------------------------------------------------------------------
command_type:COMMAND_TYPE_COMPLETE_WORKFLOW_EXECUTION  complete_workflow_execution_command_attributes:{}
2024-06-13T12:34:06.701-0700	info	dropped database	{"database": "test_0613193358_rge", "logging-call-at": "test.go:191"}
--- PASS: TestFunctionalSuite (8.28s)
    --- PASS: TestFunctionalSuite/TestDeleteWorkflowExecution_CompetedWorkflow (4.45s)
PASS
ok  	go.temporal.io/server/tests	8.874s
```

### CODE REVIEW

File: service/history/handler.go [hunk 1]
```diff
@@ -1136,6 +1136,11 @@ func (h *Handler) DeleteWorkflowExecution(ctx context.Context, request *historys
 
 	workflowExecution := request.WorkflowExecution
 	workflowID := workflowExecution.GetWorkflowId()
+	h.logger.Debug("DeleteWorkflowExecution requested",
```
Comment (yiminc) [inline service/history/handler.go:1139]: Use Info level instead of debug level. It is unusual to call API to delete a workflow.

Comment (yiminc) [reply service/history/handler.go:1139]: And I would move to after the 2 error condition check before the engine.DeleteWorkflowExecution() call.

Comment (lina-temporal) [reply service/history/handler.go]: thanks, updated!

File: service/history/handler.go [hunk 2]
```diff
@@ -1145,6 +1145,11 @@ func (h *Handler) DeleteWorkflowExecution(ctx context.Context, request *historys
 		return nil, h.convertError(err)
 	}
 
+	h.logger.Info("DeleteWorkflowExecution requested",
```
Comment (yycptt) [inline service/history/handler.go:1148]: Let's add the same thing for ForceDeleteWorkflowExecution as well.

**Review by yycptt — APPROVED** [review]

**Pull request merged**